addons/sale_timesheet/models/project_update.py

```diff
@@ -13,55 +13,66 @@ class ProjectUpdate(models.Model):
     def _get_template_values(self, project):
         return {
             **super(ProjectUpdate, self)._get_template_values(project),
+            'services': self._get_services_values(project),
             'profitability': self._get_profitability_values(project),
         }
 
+    @api.model
+    def _get_project_sols(self, project):
+        return self.env['sale.order.line'].search([('order_id', '=', project.sale_order_id.id), ('is_service', '=', True)])
+
+    @api.model
+    def _get_services_values(self, project):
+        if not project.allow_billable:
+            return {}
+        services = []
+        total_sold_per_uom = defaultdict(float)
+        total_effective_per_uom = defaultdict(float)
+        total_sold, total_effective, total_remaining = 0, 0, 0
+        sols = self._get_project_sols(project)
+        for sol in sols:
+            #We only want to consider hours and days for this calculation
+            if sol.product_uom.category_id == self.env.company.timesheet_encode_uom_id.category_id:
+                services.append({
+                    'name': sol.name,
+                    'uom': sol.product_uom.name,
+                    'sold_value': sol.product_uom_qty,
+                    'effective_value': sol.qty_delivered,
+                    'remaining_value': sol.product_uom_qty - sol.qty_delivered,
+                    'sol': sol,
+                })
+                total_sold_per_uom[sol.product_uom] += sol.product_uom_qty
+                total_effective_per_uom[sol.product_uom] += sol.qty_delivered
+        total_sold = sum([
+            uom._compute_quantity(sol.product_uom_qty, self.env.company.timesheet_encode_uom_id, raise_if_failure=False)
+            for uom, quantity in total_sold_per_uom.items()
+        ])
+        total_effective = sum([
+            uom._compute_quantity(sol.product_uom_qty, self.env.company.timesheet_encode_uom_id, raise_if_failure=False)
+            for uom, quantity in total_effective_per_uom.items()
+        ])
```

Comment:
Just so you know (I've just learnt it), we can give a list comprehension as a parameter for the sum, we can remember it for a next time

```suggestion
        total_sold = sum(
            uom._compute_quantity(sol.product_uom_qty, self.env.company.timesheet_encode_uom_id, raise_if_failure=False)
            for uom, quantity in total_sold_per_uom.items()
        )
        total_effective = sum(
            uom._compute_quantity(sol.product_uom_qty, self.env.company.timesheet_encode_uom_id, raise_if_failure=False)
            for uom, quantity in total_effective_per_uom.items()
        )
```